### PR TITLE
chore(CSharp):  added linear search algorithm

### PR DESCRIPTION
**Pull request by izzykreie**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [x] Refactor
- [x] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Added a new linear search algorithm (C#). Created unit tests for the algorithm and updated the README to reflect the newly added algorithm

**Other information:**

### CODE REVIEW

File: algorithms/CSharp/src/Search/linear-search.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Search/linear-search.cs:3]: Please remove unused `using`

File: algorithms/CSharp/src/Search/linear-search.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        public static void Main()
+        {
+            int[] numbers = { 1, 2, 3, 4, 5 };
+            int searchNumber = 3;
+            int indexOfSearchNumber = Search(numbers, searchNumber);
+
+            if (indexOfSearchNumber == -1)
+                Console.WriteLine("The number you are searching for does not exist in this list.");
+            else
+                Console.WriteLine("The number you are searching for is at index: " + indexOfSearchNumber);
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Search/linear-search.cs:18]: Please add the missing curly bracket
```suggestion
            if (indexOfSearchNumber == -1)
            {
                Console.WriteLine("The number you are searching for does not exist in this list.");
            }
            else
            {
                Console.WriteLine("The number you are searching for is at index: " + indexOfSearchNumber);
            }
```

File: algorithms/CSharp/src/Search/linear-search.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        public static int Search(int[] numbers, int searchNumber)
+        {
+            for(int index = 0; index < numbers.Length; index++)
+            {
+                if (numbers[index] == searchNumber)
+                    return index;
+            }
+            return -1;
```
Comment (ming-tsai) [inline algorithms/CSharp/src/Search/linear-search.cs:27]: Please add missing curly bracket and ensure only one `return` on this method

File: algorithms/CSharp/test/Search/linear-search.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (ming-tsai) [inline algorithms/CSharp/test/Search/linear-search.cs:3]: Please remove unused `using`

File: algorithms/CSharp/test/Search/linear-search.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+{
+    [TestFixture]
+    class LinearSearchTest
+    {
+        int[] numbers = { 1, 2, 3, 4, 5 };
+
+        [Test]
+        [TestCase(4, 3)]
+        [TestCase(5, 4)]
+        public void Should_Return_Index_Of_Matching_Element_In_Array(int searchNumber, int expectedIndexOfSearchTerm)
```
Comment (ming-tsai) [inline algorithms/CSharp/test/Search/linear-search.cs:17]: The method name is `InPutCodition_OutPutCodition` example `PassSearchNumber_ShouldGetExpectedIndex`

File: algorithms/CSharp/test/Search/linear-search.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+using Algorithms.Search;
+
+namespace Algorithms.Tests.Search
+{
+    [TestFixture]
+    class LinearSearchTest
+    {
+        int[] numbers = { 1, 2, 3, 4, 5 };
+
+        [Test]
```
Comment (ming-tsai) [inline algorithms/CSharp/test/Search/linear-search.cs:14]: This attribute is not necessarry

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @izzykreie,
I have add some comments, please check it out and let me know when you finished

**Comment by izzykreie** [comment]
@ming-tsai, thanks for reviewing my pull request! I made all of the requested changes. Please let me know if I need to make further adjustments to the code.

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**